src/Domain/JacobianDiagnostic.cpp

```diff
@@ -0,0 +1,94 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/JacobianDiagnostic.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain {
+template <size_t Dim, typename Fr>
+void jacobian_diagnostic(
+    const gsl::not_null<
+        tnsr::i<DataVector, Dim, typename Frame::ElementLogical>*>
+        jacobian_diag,
+    const Jacobian<DataVector, Dim, typename Frame::ElementLogical, Fr>&
+        analytic_jacobian,
+    const TensorMetafunctions::prepend_spatial_index<
+        tnsr::I<DataVector, Dim, Fr>, Dim, UpLo::Lo,
+        typename Frame::ElementLogical>& numeric_jacobian_transpose) {
+  const size_t number_of_points = analytic_jacobian.begin()->size();
+  destructive_resize_components(jacobian_diag, number_of_points);
+  Scalar<DataVector> jacobian_component_sum{number_of_points, 0.0};
+
+  // i_hat = logical frame index
+  // i = mapped frame index
+  for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {
+    for (size_t i = 0; i < Dim; ++i) {
```

Comment:
I think since will almost certainly use this we want to optimize a bit further:
```
jacobian_diag->get(i_hat) = -abs(abs(analytic_jacobian.get(0, i_hat));
for (size_t i = 1; ...) { ... }
get(jacobian_component_sum) =  abs(numeric_jacobian_transpose.get(i_hat, 0));
for (size_t i = 1; ...) { ... }
```
and then we can also eliminate the `0.0` on line 28, on 37  and on 44.

Reply:
Done, squashed, and force-pushed...good suggestion!